### PR TITLE
FOGL-3578: System test for foglamp-north-gcp gateway 

### PR DESCRIPTION
**Pull request by YashTatkondawar**

### CODE REVIEW

File: tests/system/python/packages/test_gcp_gateway.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+import subprocess
+import http.client
+import json
+import time
+import pytest
+from pathlib import Path
+import utils
+
+__author__ = "Yash Tatkondawar"
+__copyright__ = "Copyright (c) 2019 Dianomic Systems Inc."
```
Comment (MarkRiddoch) [inline tests/system/python/packages/test_gcp_gateway.py]: Should be 2020!

File: tests/system/python/packages/test_gcp_gateway.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+        copy_certs()
+
+        payload = {"name": task_name,
+                   "plugin": "{}".format(north_plugin),
+                   "type": "north",
+                   "schedule_type": 3,
+                   "schedule_repeat": 5,
+                   "schedule_enabled": True,
+                   "config": {"project_id": {"value": "foglamp"},
```
Comment (MarkRiddoch) [inline tests/system/python/packages/test_gcp_gateway.py]: We should probably have some notes at least as to the GCP setup we are assuming and the certs etc.

**Review by MarkRiddoch — CHANGES_REQUESTED** [review]
There is no check that data actually gets to GCP, only that we have statistics to say we sent it. Can we investigate if there is a simple way to retrieve a statistic count or something of the kind from GCP itself so that we can be more confident that the data was seen by GCP.

File: tests/system/python/packages/test_gcp_gateway.py (hunk 2 above)
Comment (YashTatkondawar) [reply tests/system/python/packages/test_gcp_gateway.py]: I have added this as [prerequisite](https://github.com/foglamp/FogLAMP/pull/1804/files#diff-2cae579d734ab10093cc71381e01aff2R9). Do you want me to add anywhere else?

**Review by MarkRiddoch — DISMISSED** [review]

File: tests/system/python/packages/test_gcp_gateway.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+        assert False, "remove package script failed!"
+
+    try:
+        subprocess.run(["cd {}/tests/system/python/scripts/package/ && ./setup {}"
+                       .format(PROJECT_ROOT, package_build_version)], shell=True, check=True)
+    except subprocess.CalledProcessError:
+        assert False, "setup package script failed"
+
+    try:
+        subprocess.run(["sudo apt install -y foglamp-north-gcp-gateway foglamp-south-sinusoid"], shell=True, check=True)
```
Comment (praveen-garg) [inline tests/system/python/packages/test_gcp_gateway.py]: We should make it `foglamp-north-gcp` from `nightly/buster/armv7l/` (atleast as long as we have it with foglamp/ org; and then think again once its migrated....)

**Review by praveen-garg — DISMISSED** [review]

**Review by praveen-garg — APPROVED** [review]

**Review by MarkRiddoch — APPROVED** [review]

**Pull request merged**